lib/assets/db_connection/main.css

```diff
@@ -79,6 +79,10 @@ input[type=number] {
   min-width: 150px;
 }
 
+.input--xs:is(input[type="text"]) {
```

Comment:
I think we should have a separate class that matches the expected behaviour, changing a generic class for specific elements can be quite confusing :)

Sidenote: [:is support](https://caniuse.com/?search=is) is fairly recent.